src/goto-instrument/code_contracts.cpp

```diff
@@ -242,6 +262,82 @@ void code_contractst::code_contracts(
       apply_contract(goto_function.body, it);
 }
 
+goto_programt code_contractst::make_ensures_assertion_for_call(
+  const code_function_callt &call)
+{
+  goto_programt goto_assertion;
+
+  // we don't handle function pointers
+  if(call.function().id() != ID_symbol)
+    return goto_assertion;
+
+  const irep_idt &function = to_symbol_expr(call.function()).get_identifier();
+  const symbolt &f_sym = ns.lookup(function);
+  const code_typet &type = to_code_type(f_sym.type);
+
+  exprt ensures = static_cast<const exprt &>(type.find(ID_C_spec_ensures));
+
+  // is there a contract?
+  if(ensures.is_nil())
+    return goto_assertion;
+
+  // replace formal parameters by arguments, replace return
+  replace_symbolt replace;
+
+  // TODO: return value could be nil
+  if(type.return_type() != empty_typet())
+  {
+    symbol_exprt ret_val(CPROVER_PREFIX "return_value", call.lhs().type());
+    replace.insert(ret_val, call.lhs());
+  }
+
+  // formal parameters
+  code_function_callt::argumentst::const_iterator a_it =
+    call.arguments().begin();
+  for(code_typet::parameterst::const_iterator p_it = type.parameters().begin();
+      p_it != type.parameters().end() && a_it != call.arguments().end();
```

Comment:
`arguments` ending before `parameters` would be a static error AFAIK -- should probably just `INVARIANT` that you have at least enough arguments (and decide what you want to do for varargs?)